### PR TITLE
Update Package Manager plugin to v1.6

### LINKED ISSUE #493: re: Homebrew Cask Updater and Homebrew Updates
FAO: (@srdgh)
FAO: (@killercup)

I am using =these two plugins together, and it seems that when I am doing a Refresh All one of the two fails as both are trying to do a `brew update` which is not possible.

Would it be possible to add in one (or both) of these plugins a condition which waits for the `brew update` to finish and than continues?

Thanks,

Rainer

### PR DESCRIPTION
**Pull request by kdeldycke**
Changelog:
* Work around the lacks of full pip upgrade command.
* Fix UnicodeDecodeError on parsing CLI output (closes #493).

### CODE REVIEW

File: Dev/Homebrew/package_manager.7h.py [hunk 1]
```diff
@@ -93,12 +110,12 @@ def active(self):
     def run(self, *args):
         """ Run a shell command, return the output and keep error message.
         """
-        self.error = None
```
Comment (tresni) [inline Dev/Homebrew/package_manager.7h.py]: Pretty sure you want to leave this as nothing else clears the error state. So without this don't you end up with a single error never going away? (Mobile so maybe I'm missing a similar line somewhere else.)

Comment (kdeldycke) [inline Dev/Homebrew/package_manager.7h.py]: Oh you're right. Overlooked this change. Going to fix it right away.

Comment (kdeldycke) [inline Dev/Homebrew/package_manager.7h.py]: Fixed in f27306433545501e2a050afa46924f70596ed19c .

**Comment by kdeldycke** [comment]
Thanks @tresni for the merge! :)

**Comment by tresni** [comment]
Sadly this fails if you have more then a handful of packages that need updated.  The issue is that BitBar only accepts [5 params](https://github.com/matryer/bitbar/blob/master/App/BitBar/Plugin.m#L220-L224).  In my case, I had 20 packages that needed updated.  It'll update the first 3 (since `install` and `--upgrade` take up `params1=` and `params2=`.)

I have an idea for a potential way to address this, hopefully have something done tonight.

**Comment by kdeldycke** [comment]
Thanks for the hint. Didn't knew about this limitation, but suspected it.

Pip full upgrade is still a hack. Having the upgrade pass limited to 5 items is good enough to me. It's just slightly inconvenient as you have to call it a couple of time to flush the whole queue. But the plugin is there to remind you about the left-over upgrades.

Good if you find a better workaround but that's not a high-priority task.

**Comment by tresni** [comment]
I think I have a decent work around that will allow us to support any package manager that doesn't normally support a full update. PR should be coming this weekend, but I am traveling so may be delayed ;)

-- Brian

**Pull request merged**